### PR TITLE
hotfix/CP-4614-unsubscribe-after-channel-id-change-not-working-push-r…

### PR DESCRIPTION
**Pull request by jprajapati1996**
hotfix/CP-4614 Unsubscribe after channelId change not working

 Push Received Crash issue

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java [hunk 1]
```diff
@@ -87,6 +87,9 @@ public static void process(Context context, Notification notification, Subscript
         }
 
         try {
+            if (maximumNotifications <= 0) {
+                return;
+            }
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java:92]: Add new line after

File: cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java [hunk 2]
```diff
@@ -108,9 +111,11 @@ public static void process(Context context, Notification notification, Subscript
                 notifications = new LimitedSizeQueue<>();
             }
             notifications.setCapacity(maximumNotifications);
-
             if (notification.getCreatedAt() == null || notification.getCreatedAt().equalsIgnoreCase("")) {
-                String currentDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX", Locale.US).format(new Date());
+                String currentDate = null;
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java:116]: Will this prevent a crash? And can we still define `currentDate` to any value for Android < 7 ?

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java (hunk 2 above)
Comment (gbhrdt) [reply cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java:116]: @jprajapati1996

File: cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java (hunk 1 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java:92]: Added

File: cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java (hunk 2 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java:116]:
```
            if (notification.getCreatedAt() == null || notification.getCreatedAt().equalsIgnoreCase("")) {
                String currentDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX", Locale.US).format(new Date());
                notification.setCreatedAt(currentDate);
            }

```

**Pull request merged**